pkgs/stdenv/adapters.nix

```diff
@@ -245,6 +246,11 @@ rec {
          * wrapper instead? */
         cc = targetStdenv.cc.cc;
 
+        /* NOTE: We must provide coreutils explicitly so that wrapCCWith
+         * doesn't take it from pkgs, which is pkgsHostTarget, because it
+         * would fail to run on the buildPlatform when cross-compiling. */
+        coreutils = targetStdenv.cc.coreutils_bin;
```

Comment:
It just occurred to me that instead of calling `wrapCCWith` directly we could assume that `targetStdenv.cc` is already a wrapper, and then we could take `targetStdenv.cc.override` with the same arguments

Reply:
It's not a great assumption but it might be better than what we do now

Reply:
Btw note https://github.com/NixOS/nixpkgs/issues/283517 if you haven't seen it yet

Reply:
Mhm, good observation about avoiding roundtrip wrapping. I'm curious if it'll break anything in Nixpkgs as I've implemented it -- it looks like `pythonPackages.tensorflow`, `gnuradio3_8`, and `cudaPackages.backendStdenv` are the only users within Nixpkgs.

I had seen the issue you opened -- I agree that there needs to be a better interface, but for the very short term work I have over the next two weeks, I think it's out of scope and I'd rather focus on this aspect of it. Does that make sense?

Reply:
> I'm curious if it'll break anything in Nixpkgs as I've implemented it

Within Nixpkgs it should break strictly less (e.g. the previous implementation definitely could not be compatible with various `nixpkgsFun` instances like `pkgsStatic`, because it was just discarding the old wrapper's information). The exact interaction of `useLibsFrom` and custom `wrapCCWith` invocations, and the impact out of tree are still unclear.

> I agree that there needs to be a better interface, but for the very short term work I have over the next two weeks

I agree with switching to `.override` now, but I'd like to emphasize that the issue is relevant in the short term (1mo) too because it'll be harder to delete the adapter after `24.05`